### ISSUE TITLE
Bitfinex Wrong account value multiple instances, single account

### ISSUE THREAD

**Issue by CooleRnax** [ctubio/Krypto-trading-bot#880]
When you run multiple instances with different pairs on single bitfinex account total account value is calculated wrongly.

Please fix this.

**Comment by ctubio** [comment]
i can try to fix this even if the description is the smallest ever seen (no logs, no example values)

Please understand this: that what is wrong for you may be excelent for me, or for my sister (that is always against my opinion)

**Comment by ctubio** [comment]
consider to share what multiple currency paris are you using, so i can reproduce your very same issue

**Comment by ctubio** [comment]
:koala: and the exchange name please

**Comment by ctubio** [comment]
he said bitfinex!

**Comment by CooleRnax** [comment]
example 10+ pairs on single bitfinex account all trade vs usd.
![image](https://user-images.githubusercontent.com/980180/53284227-555d2080-3762-11e9-9c82-cc96643c0786.png)
![image](https://user-images.githubusercontent.com/980180/53284234-64dc6980-3762-11e9-9978-6ce34f85b2b1.png)

**Comment by ctubio** [comment]
:koala: ok thanks for the info

**Comment by ctubio** [comment]
please note the calucaltion of wallet values is a very simple formula:
https://github.com/ctubio/Krypto-trading-bot/blob/79f0971ea5682e45d84a838b5210362c3e9ea824/src/lib/Krypto.ninja-data.h#L1812-L1815

(it can't be wrong)

the amounts that you are missing are held in other markets.

in other words:
DAI/USD wallet misses the held amounts in open orders of both sides on TRX/USD, XLM/USD and ZRX/USD
TRX/USD wallet misses the held amounts in open orders of both sides on DAI/USD, XLM/USD and ZRX/USD
XLM/USD wallet misses the held amounts in open orders of both sides on TRX/USD, DAI/USD and ZRX/USD
ZRX/USD wallet misses the held amounts in open orders of both sides on TRX/USD, XLM/USD and DAI/USD

in other words:
bitfinex website knows all your funds held and available in all your wallets
each K bot only knows about the funds held and available in its own wallet/market

hope this helps'''

**Comment by CooleRnax** [comment]
i think this approach only works if you use single bot on one account
if you want to scale and use % values everything will start working not as it should

in my example you start with 1000 usd, 0 coins and 10 bots.
starting account value per bot is 1000. max order value is 3%
bots begin sending orders **30$** each 3%
lets assume that all bots bought successfully.
new account values for bots is 1000-(30*9)=**730$**
second wave of orders starts. bots begin sending orders  **21.9$** each 3%
lets assume that all bots bought successfully again.
new account values for bots is 730-(21.9*9)=**532$**
second wave of orders starts. bots begin sending order **15.9$** each 3%

same problems will have every setting where you use % (positions, orders), except ping pong width size
if everything is connected to account value, it should help

**Comment by ctubio** [comment]
you are right but that is expected; K intentionaly doesnt read all wallets from an exchange; just the currency begin traded, because happens the same when manually trading, you cannot place 1000$ in 4 markets, you can only place 250€ on each (if our world had always an equitable distribution of resources)
# ☭

**Comment by ctubio** [comment]
(really the reason why this is not implemented is *performance*)

**Comment by CooleRnax** [comment]
maybe you can include option to override this by my own value?

**Comment by CooleRnax** [comment]
something like max account value usage in usd

**Comment by CooleRnax** [comment]
bot will think that 100 is max for account and will not buy more

**Comment by CooleRnax** [comment]
example of the problem i'm facing right now:
![image](https://user-images.githubusercontent.com/980180/53414850-c447ad00-39e0-11e9-9cef-36a93a13c02e.png)
one instance overbought one coin, and all other instances have lover account values and pefroming worce because of this

**Comment by CooleRnax** [comment]
ideally it would be great to add max account usage in %
this variable will be the value of the account
and all other settings will not requre updates

**Comment by ctubio** [comment]
mm.. i think you mean argument --wallet-limit, see the --help:
https://github.com/ctubio/Krypto-trading-bot/blob/048f945eafeef9420c86573674634211103c9efa/src/bin/trading-bot/trading-bot.h#L32-L33

you can set then a maximum limit value for each bot, that K will use as a maximum wallet (ignoring the rest of funds)

**Comment by CooleRnax** [comment]
this requires restart of the bot which is not very practical

**Comment by CooleRnax** [comment]
wallet-limit is not ideal but will make my life much easier for running multiple instances.
where i can find all possible -- commands?
maybe i can find a solution for some other issues there...

**Comment by CooleRnax** [comment]
if symbol is btcusd, wallet-limit will limit usage in usd or in btc?

**Comment by CooleRnax** [comment]
i have tried --wallet-limit 250
in the result account value is still bigger than 250 usd, same as was before
the actual behavior haven't changed
all orders and positions are still linked to account value

**Comment by CooleRnax** [comment]
--wallet-limit works for btc in btcusd pair
I need something to set max account value in usd currency.

**Comment by CooleRnax** [comment]
started dedicated issue
https://github.com/ctubio/Krypto-trading-bot/issues/883

**Comment by ctubio** [comment]
--wallet-limit uses base currency, see --help
